Moonlight macOS/Input/ControllerSupport.m

```diff
@@ -0,0 +1,63 @@
+//
+//  ControllerSupport.m
+//  Moonlight macOS
+//
+//  Created by Felix Kratz on 15.03.18.
+//  Copyright © 2018 Felix Kratz. All rights reserved.
+//
+
+#import "ControllerSupport.h"
+#import "DataManager.h"
+#include "Limelight.h"
+#include "Gamepad.h"
+
+@class Controller;
+
+@implementation ControllerSupport {
+    NSLock *_controllerStreamLock;
+    NSMutableDictionary *_controllers;
+    char _controllerNumbers;
+    Controller* _controller;
+}
+
+-(void) updateButtonFlags:(Controller*)controller flags:(int)flags
+{
+    @synchronized(controller) {
+        controller.lastButtonFlags = flags;
+    }
+}
+
+-(void) setButtonFlag:(Controller*)controller flags:(int)flags
+{
+    @synchronized(controller) {
+        controller.lastButtonFlags |= flags;
+    }
+}
+
+-(void) clearButtonFlag:(Controller*)controller flags:(int)flags
+{
+    @synchronized(controller) {
+        controller.lastButtonFlags &= ~flags;
+    }
+}
+
+-(void) updateFinished:(Controller*)controller
+{
+    _controllerNumbers = 0;
```

Comment:
If you go for multi-controller support, I'd make `updateFinished` take the active controller bitmask as a parameter, since you won't have easy access to it here.